docs/_docs/02_features/provider-cache.md

```diff
@@ -0,0 +1,132 @@
+
+### Usage
+
+Terragrunt Provider Cache is currently considered an experimental feature, so it is disabled by default. To enable it you need to use the flag [`terragrunt-provider-cache`](https://terragrunt.gruntwork.io/docs/reference/cli-options/#terragrunt-provider-cache):
+
+``` shell
+terragrunt run-all apply --terragrunt-provider-cache
+```
+or the environment variable `TERRAGRUNT_PROVIDER_CACHE`:
+
+``` shell
+TERRAGRUNT_PROVIDER_CACHE=1 terragrunt run-all apply
+```
+
+By default, cached providers are stored in `terragrunt/providers` folder, which is located in the user cache directory:
+
+* `$HOME/.cache/terragrunt/providers` on Unix systems
+* `$HOME/Library/Caches/terragrunt/providers` on Darwin
+* `%LocalAppData%\terragrunt\providers` on Windows
+
+The file structure of the cache directory is identical to the Terraform [plugin_cache_dir](https://developer.hashicorp.com/terraform/cli/config/config-file#provider-plugin-cache) directory. If you already have a directory with providers cached by Terraform [plugin_cache_dir](https://developer.hashicorp.com/terraform/cli/config/config-file#provider-plugin-cache), you can set this path using the flag [`terragrunt-provider-cache-dir`](https://terragrunt.gruntwork.io/docs /link/cli-options/#terragrunt-provider-cache-dir), to make cache server reuse them.
+
+``` shell
+terragrunt plan \
+--terragrunt-provider-cache \
+--terragrunt-provider-cache-dir /new/path/to/cache/dir
+```
+
+or the environment variable `TERRAGRUNT_PROVIDER_CACHE_DIR`:
+
+``` shell
+TERRAGRUNT_PROVIDER_CACHE=1 \
+TERRAGRUNT_PROVIDER_CACHE_DIR=/new/path/to/cache/dir \
+terragrunt plan
+```
+
+By default, Terragrunt only caches providers from the following registries: `registry.terraform.io`, `registry.opentofu.org`. You can override this list using the flag [`terragrunt-provider-cache-registry-names`](https://terragrunt.gruntwork.io/docs/reference/cli-options/#terragrunt-provider-cache-registry-names):
+
+
+``` shell
+terragrunt apply \
+--terragrunt-provider-cache \
+--terragrunt-provider-cache-registry-names example1.com \
+--terragrunt-provider-cache-registry-names example2.com
+```
+
+or the environment variable `TERRAGRUNT_PROVIDER_CACHE_REGISTRY_NAMES`:
+
+``` shell
+TERRAGRUNT_PROVIDER_CACHE=1 \
+TERRAGRUNT_PROVIDER_CACHE_REGISTRY_NAMES=example1.com,example2.com \
+terragrunt apply
+```
+
+Since Terragrunt Cache is essentially a Private Registry server that accepts requests from Terraform, downloads and saves providers to the cache directory, there are a few more flags that are unlikely to be needed, but are useful to know about:
```

Comment:
I'd move all these docs under a new "Configure the Terragrunt Cache Provider" sub-section (`###` heading) that goes _below_ the "How Terragrunt Provider Caching works" section, as until you read that, it's not clear why there's any server at all!

Reply:
I did as you said. Thank you.